### PR TITLE
verify dropwizard application log

### PR DESCRIPTION
**Pull request by sullis**
# Description

add test to verify that there are zero errors in the dropwizard application log

## Type of change

- [x] test suite

# How Has This Been Tested?

Github Actions CI

# Checklist:

- [x] I have performed a self-review of my code

### CODE REVIEW

File: polaris-service/src/test/java/org/apache/polaris/service/PolarisApplicationIntegrationTest.java [hunk 1]
```diff
@@ -179,6 +203,14 @@ public static void deletePrincipalRole() {
         .close();
   }
 
+  private static void assertZeroErrorsInApplicationLog() {
+    assertThat(new File(CURRENT_LOG.get()))
+        .exists()
+        .content()
+        .hasSizeGreaterThan(0)
+        .doesNotContain("ERROR", "FATAL", "Exception");
```
Comment (flyrain) [inline polaris-service/src/test/java/org/apache/polaris/service/PolarisApplicationIntegrationTest.java:211]: I think a warn-level log could also have `Exception`, which shouldn't be considered as an error.

Comment (sullis) [reply polaris-service/src/test/java/org/apache/polaris/service/PolarisApplicationIntegrationTest.java]: Fixed.

**Comment by sullis** [comment]
ready for review

File: polaris-service/src/test/java/org/apache/polaris/service/PolarisApplicationIntegrationTest.java [hunk 2]
```diff
@@ -103,6 +106,11 @@
   SnowmanCredentialsExtension.class
 })
 public class PolarisApplicationIntegrationTest {
+  @TempDir private static Path tempDir;
+  private static Supplier<String> CURRENT_LOG = () -> tempDir.resolve("application.log").toString();
+  private static Supplier<String> ARCHIVED_LOG =
+      () -> tempDir.resolve("application-%d-%i.log.gz").toString();
```
Comment (flyrain) [inline polaris-service/src/test/java/org/apache/polaris/service/PolarisApplicationIntegrationTest.java:112]: Looks like the archive logs are not used anywhere. Do we need this change?

File: polaris-service/src/test/java/org/apache/polaris/service/PolarisApplicationIntegrationTest.java [hunk 3]
```diff
@@ -103,6 +106,11 @@
   SnowmanCredentialsExtension.class
 })
 public class PolarisApplicationIntegrationTest {
+  @TempDir private static Path tempDir;
+  private static Supplier<String> CURRENT_LOG = () -> tempDir.resolve("application.log").toString();
```
Comment (flyrain) [inline polaris-service/src/test/java/org/apache/polaris/service/PolarisApplicationIntegrationTest.java:110]: Can we make it a final field?

File: polaris-service/src/test/java/org/apache/polaris/service/PolarisApplicationIntegrationTest.java [hunk 4]
```diff
@@ -115,7 +123,11 @@ public class PolarisApplicationIntegrationTest {
               "server.applicationConnectors[0].port",
               "0"), // Bind to random port to support parallelism
           ConfigOverride.config(
-              "server.adminConnectors[0].port", "0")); // Bind to random port to support parallelism
+              "server.adminConnectors[0].port", "0"), // Bind to random port to support parallelism
+          ConfigOverride.config("logging.appenders[1].type", "file"),
+          ConfigOverride.config("logging.appenders[1].currentLogFilename", CURRENT_LOG),
+          ConfigOverride.config("logging.appenders[1].archivedLogFilenamePattern", ARCHIVED_LOG),
+          ConfigOverride.config("logging.appenders[1].maxFileSize", "2000000"));
```
Comment (flyrain) [inline polaris-service/src/test/java/org/apache/polaris/service/PolarisApplicationIntegrationTest.java:130]: Just curious, is 2M big enough? Another option is to check archived log as well.

File: polaris-service/src/test/java/org/apache/polaris/service/PolarisApplicationIntegrationTest.java [hunk 5]
```diff
@@ -179,6 +198,14 @@ public static void deletePrincipalRole() {
         .close();
   }
 
+  private static void assertZeroErrorsInApplicationLog() {
```
Comment (flyrain) [inline polaris-service/src/test/java/org/apache/polaris/service/PolarisApplicationIntegrationTest.java:201]: Do we need to check archived log file as well?

**Review by flyrain — COMMENTED** [review]
Thanks @sullis for working on it. LGTM overall. Left minor comments and some questions. The only thing I'm not sure is whether we need to check the archived log. +1 if 2M is big enough for the log in the test.

File: polaris-service/src/test/java/org/apache/polaris/service/PolarisApplicationIntegrationTest.java (hunk 2 above)
Comment (sullis) [reply polaris-service/src/test/java/org/apache/polaris/service/PolarisApplicationIntegrationTest.java]: I just cleaned this up.  We don't need 'archivedLogFilenamePattern' or ARCHIVED_LOG

File: polaris-service/src/test/java/org/apache/polaris/service/PolarisApplicationIntegrationTest.java (hunk 3 above)
Comment (sullis) [reply polaris-service/src/test/java/org/apache/polaris/service/PolarisApplicationIntegrationTest.java]: Done.

File: polaris-service/src/test/java/org/apache/polaris/service/PolarisApplicationIntegrationTest.java (hunk 4 above)
Comment (sullis) [reply polaris-service/src/test/java/org/apache/polaris/service/PolarisApplicationIntegrationTest.java]: maxFileSize is not required. I removed it.

File: polaris-service/src/test/java/org/apache/polaris/service/PolarisApplicationIntegrationTest.java (hunk 5 above)
Comment (sullis) [reply polaris-service/src/test/java/org/apache/polaris/service/PolarisApplicationIntegrationTest.java:199]: We don't.  I removed the Archived Log config.

**Comment by sullis** [comment]
Ready for review.

**Review by collado-mike — APPROVED** [review]
Slick

**Review by adutra — APPROVED** [review]

**Comment by sullis** [comment]
waiting for GitHub Actions approval.

**Comment by sullis** [comment]
GitHub workflows awaiting approval.

**Comment by sullis** [comment]
workflows awaiting approval

**Comment by flyrain** [comment]
@sullis, looks like there is a conflict. It'd be nice if you can fix it, so that we can merge it.

**Comment by sullis** [comment]
Workflows waiting for approval

**Comment by sullis** [comment]
GitHub Actions Workflows waiting for approval

**Review by adutra — APPROVED** [review]
